src/components/structures/login/Registration.js

```diff
@@ -135,154 +124,124 @@ module.exports = React.createClass({
-                    self.props.referrer || '', // Default to empty string = not referred
-                    self.registerLogic.params.idSid,
-                    self.registerLogic.params.clientSecret
-                ).then((data) => {
-                    const teamToken = data.team_token;
-                    // Store for use /w welcome pages
-                    window.localStorage.setItem('mx_team_token', teamToken);
-
-                    self._rtsClient.getTeam(teamToken).then((team) => {
-                        console.log(
-                            `User successfully registered with team ${team.name}`
-                        );
-                        if (!team.rooms) {
-                            return;
+        // Done regardless of `teamSelected`. People registering with non-team emails
+        // will just nop. The point of this being we might not have the email address
+        // that the user registered with at this stage (depending on whether this
+        // is the client they initiated registration).
+        let trackPromise = q(null);
+        if (this._rtsClient) {
+            // Track referral if this.props.referrer set, get team_token in order to
+            // retrieve team config and see welcome page etc.
+            trackPromise = this._rtsClient.trackReferral(
+                this.props.referrer || '', // Default to empty string = not referred
+                this.registerLogic.params.idSid,
+                this.registerLogic.params.clientSecret
+            ).then((data) => {
+                const teamToken = data.team_token;
+                // Store for use /w welcome pages
+                window.localStorage.setItem('mx_team_token', teamToken);
+                this.props.onTeamMemberRegistered(teamToken);
+
+                this._rtsClient.getTeam(teamToken).then((team) => {
+                    console.log(
+                        `User successfully registered with team ${team.name}`
+                    );
+                    if (!team.rooms) {
+                        return;
+                    }
+                    // Auto-join rooms
+                    team.rooms.forEach((room) => {
+                        if (room.auto_join && room.room_id) {
+                            console.log(`Auto-joining ${room.room_id}`);
+                            MatrixClientPeg.get().joinRoom(room.room_id);
                         }
-                        // Auto-join rooms
-                        team.rooms.forEach((room) => {
-                            if (room.auto_join && room.room_id) {
-                                console.log(`Auto-joining ${room.room_id}`);
-                                MatrixClientPeg.get().joinRoom(room.room_id);
-                            }
-                        });
-                    }, (err) => {
-                        console.error('Error getting team config', err);
                     });
 
                     return teamToken;
```

Comment:
Now that I've got github to give me a sensible diff, I can see you've moved this inside the `getTeam.then` callback - previously `teamToken` would be returned even if the `getTeam` call failed. Was that deliberate? I suspect it will be fine, but thought I'd check.